veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,586 @@
+          hypervisorDevice: kvm
+          virtType: kvm
+        developerConfiguration:
+          featureGates:
+            - ConfigurableHypervisor
+    ```
+
+- `virt-controller` reads the configured hypervisor from `ClusterConfig` when generating launcher manifests and threads that ID through the `ConverterContext` so downstream components can act consistently.
+- Each package's registry uses the configured name to locate its implementation, avoiding a monolithic factory while keeping selection logic consistent.
+
+### Integration with Defaults, Runtime, Converter and Validating Webhooks
+
+1. The proposed `Validator` interface's hypervisor-specific implementation would be resolved and invoked from within the `Admit` function of the concerned `Admitter` implementations - e.g., `VMsAdmitter`, `VMICreateAdmitter`, `VMIUpdateAdmitter`, `VMIRSAdmitter`, etc.
+2. `pkg/defaults` pulls the `DefaultsExtension` associated with the configured hypervisor to mutate the VMI and set the appropriate default values for the specific hypervisor supported on the cluster.
+3. `virt-controller` reads the configured hypervisor from `ClusterConfig`, and adds a K8s device request for the appropriate hypervisor device to the `virt-launcher` pod definition so that it can be scheduled on the node with that hypervisor device. Furthermore, it adds the hypervisor information to the command-line of the `virt-launcher`, which the `virt-launcher` pod then uses to instantiate the right implementation of the `HypervisorConverter` for converting VMI spec to Libvirt domain XML.
+4. The `virt-controller` resolves the `VirtLauncherResourceRenderer` implementation to run `GetMemoryOverhead`, keeping pod-level resource calculations in sync with the mutated spec. The `DomainTuner` interface is used by virt-handler to run the `AdjustResources` function for memlock sizing and ancillary bookkeeping.
+5. When virt-launcher converts the VMI, it instantiates both the `Converter` and the Hypervisor Runtime interfaces. The hypervisor-specific instance of the `Converter` interface leverages common conversion functions as well as hypervisor-specific functions to convert VMI to Libvirt domain.
+6. The virt-launcher component will continue to leverage existing helpers for common functionality, such as setLaunchSecurity, disk configuration, and network setup. Hypervisor-specific extensions to the converter logic will extend the base implementation of these helpers to introduce specialized logic.
+
+
+### Converter Restructuring Inside Virt-launcher
+
+- The existing `pkg/virt-launcher/virtwrap/converter` package becomes a reusable library with a new `hypervisor` subpackage. Introduce a new interface named `HypervisorConverter` that exposes the main functions for converting VMI spec to Libvirt domain XML.
+
+  ```golang
+  type Converter interface {
+    Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInstance, domain *api.Domain, c *ConverterContext) (err error)
```

Comment:
As mentioned in our call yesterday, did you check that this still aligns with the recent refactorings of the converter?